app/models/hmpps_api/offender.rb

```diff
@@ -257,6 +257,15 @@ def inside_omic_policy?
         criminal_sentence? && convicted?
     end
 
+    def latest_early_allocation
+      allocation = current_sentence_allocations(sentence, @case_information&.early_allocations || [])&.last
+      allocation if allocation.present? && allocation.created_within_referral_window? && (allocation.eligible? || allocation.community_decision?)
+    end
+
+    def current_sentence_allocations(sentence, early_allocations)
+      early_allocations.select { |early_allocation| early_allocation.assessment_date.after?(sentence.sentence_start_date) }
+    end
```

Comment:
Think we can be smarter than this: something like

def early_allocations
   @case.informatiom.early_allocations.where(created_at >= ?, sentence_start_date)
end

then the controller should feed this rather than the raw data to the view - all the offender parameters go away again

Reply:
It would also be good to add a check to the early_allocations_controller :index action to ensure that it picks up all the allocations not just the pre-sentence ones - and maybe same for allocation history